### ISSUE TITLE
gatsby build failed due to missing elements in SSR

### ISSUE THREAD

**Issue by LER0ever** [narative/gatsby-theme-novela#281]
# Prerequisites
* [x] Have you searched the issue tracker for similar issues (either past or present)?
* [x] Can you reproduce the issue consistently?
* [x] Are you reporting to the correct repository?
* [ ] Is this something you can **debug and fix**? 

# Description
`yarn start` using gatsby serve works perfectly, but `yarn build` does not, because of an error from `react-medium-image-zoom` in SSR mode.

### Version of Novela you are using
master

### Number of posts you have and which data source
* [x] Local: just those from this repo
* [ ] Contentful

### Describe the bug
This `yarn build` error only shows recently after `react-medium-image-zoom` was updated to v4, i.e. **the PR #273 .**
If I **revert** the pull request #273 and the corresponding commits on top of the current master, `yarn build` will no longer produce errors, and the site deploys successfully.

Error Log while doing SSR:
```
failed Building static HTML for pages - 5.551s

 ERROR #95312

"document" is not available during server side rendering.

See our docs page for more info on this error: https://gatsby.dev/debug-html


  68 |     overlayBgColorStart: overlayBgColorStart,
  69 |     parentRef: wrapRef,
> 70 |     portalEl: portalEl || document.body,
     | ^
  71 |     scrollableEl: scrollableEl || window,
  72 |     transitionDuration: transitionDuration,
  73 |     zoomMargin: zoomMargin,


  WebpackError: ReferenceError: document is not defined

  - Controlled.js:70 Controlled
    node_modules/react-medium-image-zoom/dist/esm/Controlled.js:70:1

  - Code.Prism.tsx:36 Promise._execute
    @narative/gatsby-theme-novela/src/components/Code/Code.Prism.tsx:36:9

  - Anchor.tsx:18 MappingPromiseArray.PromiseArray._iterate
    @narative/gatsby-theme-novela/src/components/Anchor/Anchor.tsx:18:16

```

### To Reproduce
0. Clone the master branch
1. First fix #249 with a workaround: delete `gatsby-plugin-contentful` in `www/package.json` and re-run `yarn install`.
2. Now `yarn start` should work (without the error in #249 ), and the site is working fine at `localhost:8000`
3. Run `yarn build`. Should see the error provided above: `document is not defined`

### Expected behavior
`yarn build` should produce the SSR-ed static website in `public`

### Screenshots
![image](https://user-images.githubusercontent.com/10568668/74584783-cbf8cb00-5010-11ea-8444-12912ecac13c.png)

### Additional context

might be relevant:
- `react-medium-image-zoom` recently fixed SSR build in PR: https://github.com/rpearce/react-medium-image-zoom/pull/206 
- seems user from https://github.com/narative/gatsby-theme-novela/issues/277 are seeing the exact same error

**Comment by LER0ever** [comment]
I've created a draft PR #282  for POC, which deletes `gatsby-plugin-contentful` temporarily as well as reverts #273 . The automatic check for that PR 
https://github.com/narative/gatsby-theme-novela/pull/282/checks?check_run_id=447584748 
shows that it builds correctly now. 

But since I don't know much about React or Gatsby, I'm still not sure what's the root cause of this issue.

**Comment by jpvalery** [comment]
I don't think we can remove Contentful as it's a core feature of the theme (@brotzky ?) 

Regarding `react-medium-image-zoom` what was the changelog for V4?
